src/hats/pixel_math/healpix_shim.py

```diff
@@ -40,21 +40,31 @@ def order2npix(order: int) -> int:
     return 12 * (1 << (2 * order))
 
 
-def order2resol(order: int, arcmin: bool = False) -> float:
-    resol_rad = np.sqrt(order2pixarea(order))
-
+def order2resol(order: int, *, arcmin: bool = False, unit=u.rad) -> float:
     if arcmin:
-        return np.rad2deg(resol_rad) * 60
+        unit = u.arcmin
+
+    return np.sqrt(order2pixarea(order, unit=unit))
 
-    return resol_rad
 
+def order2pixarea(order: int, *, degrees: bool = False, unit=u.rad) -> float:
```

Comment:
Maybe we can use solid angle units and use astropy for unit conversion? So we start with `pix_area_ster = 4 * np.pi / npix * u.ster` and `return pix_area_rad.to_value(u.Unit(unit))`. This would also allow `unit: string` such as `arcmin**2` or `ster`

Reply:
Dig it!